### ISSUE TITLE
Pick List doesn't have auditory indicators in VoiceOver to signal selection

### ISSUE THREAD

**Issue by FelFly** [Esri/calcite-components#4434]
### Summary

The Pick List component presently relies on a visual element to indicate when an option is active and it does not communicate a change in selection through auditory indicators. This is a failure of [1.3.1 Sensory Characteristics](https://www.w3.org/WAI/WCAG21/Understanding/sensory-characteristics.html)

### Actual Behavior

When using the arrow keys and a screen-reader there is no auditory indicator that there is an active selection nor is there an auditory indicator for when the active selection has been changed. When using native VoiceOver controls, there is an auditory indicator of active selection when passively moving through elements, but the actual event of choosing a new selection does not have any auditory indicator that anything has changed.

Video: https://esriis-my.sharepoint.com/:v:/g/personal/klar7790_esri_com/Eb65u_ZciUxImiwg-nJv_5cBgfTGTqqlgS6q32LzEJTwOg

### Expected Behavior

When navigating the pick list with a screen-reader, I would expect to hear "checked/unchecked" so that I know which list item is selected. That's my first preference, because I think it communicates the most auditory information. However, even if unselected items were not announced, I would expect to hear newly selected items announce themselves as selected.

### Reproduction Sample

https://developers.arcgis.com/calcite-design-system/components/pick-list/

### Reproduction Steps

STR:
1. In Safari go to https://developers.arcgis.com/calcite-design-system/components/pick-list/
2. Turn on VoiceOver (CMD+F5)
3. Navigate down into the Pick List example
4. Using VO (Ctrl+Option) keys and right-arrow, move through list elements.
5. On Maps you should hear "selected"
6. Move to Layers, press VO+Spacebar
7. Visually, the blue bullet will move to Layers; however, you will hear nothing

Alternatively:
1. Pick up on step 3 above
2. Use only the arrow keys to move through list elements
3. On the selected "Maps" option, you will not hear "selected"
4. Down-arrow to Layers, press spacebar, you will not hear any indicator that it is selected

### Reproduction Version

1.0.0-beta.81

### Relevant Info

Safari Version 15.4 (15613.1.17.1.13)
VoiceOver (macOS Catalina, 10.15.7)

**Comment by geospatialem** [comment]
On hold with `list` component rework process.

**Comment by geospatialem** [comment]
**_Much delayed_**, but after taking a more in-depth look with the refactored `calcite-list`, unfortunately there isn't much support with the [gridcell role and `aria-selected`](https://a11ysupport.io/tests/aria_gridcell(aria-selected)#assertion-aria-aria-selected_attribute-convey_value_false-aria-gridcell_role). 

Confirmed similar behavior with both `calcite-list` in `2.12.2` with this [Calcite 2.12.2 sample](https://codepen.io/geospatialem/pen/OJeweag) and the [a11ysupport.io test](https://a11ysupport.io/tests/html/aria_gridcell(aria-selected).html). 

Marking the above as not planned, but note that the components are setup to provide support, and _some_ context is provided to some assistive technologies, such as NVDA, as of testing conducted in September 2024.
